ui/src/pages/version/components/forms/components/AutoscalingPolicyFormGroup.js

```diff
@@ -0,0 +1,137 @@
+import React, { Fragment } from "react";
+import {
+  EuiFormRow,
+  EuiFieldNumber,
+  EuiDescribedFormGroup,
+  EuiSuperSelect,
+  EuiText,
+  EuiToolTip,
+  EuiIcon
+} from "@elastic/eui";
+
+export const AutoscalingPolicyFormGroup = ({
+  deploymentMode,
+  autoscalingPolicy,
+  onAutoscalingPolicyChange
+}) => {
+  const onMetricsTypeChange = value => {
+    onAutoscalingPolicyChange({
+      metrics_type: value,
+      target_value: autoscalingPolicy.target_value
+    });
+  };
+
+  const onTargetValueChange = e => {
+    onAutoscalingPolicyChange({
+      metrics_type: autoscalingPolicy.metrics_type,
+      target_value: e.target.value
+    });
+  };
+
+  const allMetricsType = [
+    {
+      value: "cpu_utilization",
+      inputDisplay: "CPU Utilization",
+      dropdownDisplay: (
+        <Fragment>
+          <strong>CPU Utilization</strong>
+          <EuiText size="s" color="subdued">
+            <p className="euiTextColor--subdued">
+              Autoscaling based on average cpu usage
+            </p>
+          </EuiText>
+        </Fragment>
+      )
+    },
+    {
+      value: "memory_utilization",
+      inputDisplay: "Memory Utilization",
+      dropdownDisplay: (
+        <Fragment>
+          <strong>Memory Utilization</strong>
+          <EuiText size="s" color="subdued">
+            <p className="euiTextColor--subdued">
+              Autoscaling based on average memory usage
+            </p>
+          </EuiText>
+        </Fragment>
+      )
+    },
+    {
+      value: "concurrency",
+      inputDisplay: "Concurrency",
+      dropdownDisplay: (
+        <Fragment>
+          <strong>Concurrency</strong>
+          <EuiText size="s" color="subdued">
+            <p className="euiTextColor--subdued">
+              Autoscaling based on number of concurrent request being processed
+            </p>
+          </EuiText>
+        </Fragment>
+      )
+    },
+    {
+      value: "rps",
+      inputDisplay: "Request per Second",
+      dropdownDisplay: (
+        <Fragment>
+          <strong>Request per Second</strong>
+          <EuiText size="s" color="subdued">
+            <p className="euiTextColor--subdued">
+              Autoscaling based on throughput (RPS)
+            </p>
+          </EuiText>
+        </Fragment>
+      )
+    }
+  ];
+
+  const filterMetricsOptions = (allOptions, deploymentMode) => {
+    var allowedMetrics = [
+      "cpu_utilization",
+      "memory_utilization",
+      "concurrency",
+      "rps"
+    ];
+    if (deploymentMode === "raw_deployment") {
+      allowedMetrics = ["cpu_utilization"];
+    }
+
+    return allOptions.filter(option => allowedMetrics.includes(option.value));
+  };
+
+  return (
+    <EuiDescribedFormGroup
+      title={<p>Autoscaling Policy</p>}
+      description={
+        <Fragment>
+          Autoscaling Policy determine the condition for adding additional
+          replica
```

Comment:
Autoscaling Policy determines the condition for increasing or decreasing replicas.

Reply:
Updated